store/src/java/com/zimbra/cs/mailbox/Mailbox.java

```diff
@@ -9391,55 +8835,53 @@ public Chat updateChat(OperationContext octxt, ParsedMessage pm, int id) throws
         int size = (int) staged.getSize();
 
         SaveChat redoRecorder = new SaveChat(mId, id, digest, size, -1, 0, null);
-        boolean success = false;
-        try {
-            beginTransaction("saveChat", octxt, redoRecorder);
-
-            SaveChat redoPlayer = (SaveChat) currentChange().getRedoPlayer();
+        try (final MailboxLock l = lockFactory.writeLock()) {
+            final MailboxTransaction t = new MailboxTransaction("saveChat", octxt, l, redoRecorder);
```

Comment:
not in try-with-resources block

Reply:
It's because `t.close()` happens before `sm.quietDelete(staged)` and I was trying to maintain order.